includes/class-drift-main.php

```diff
@@ -0,0 +1,621 @@
+  			}
+        if ( ! empty( $input['js_hook'] ) ) {
+          $input['js_hook'] = sanitize_text_field( 'on' );
+        } else {
+          unset( $input['js_hook'] );
+        }
+      }
+      return $input;
+    }
+
+    /**
+     * Enqueues script in dashboard.
+     *
+     * Adds jQuery script in dashboard plugin settings page.
+     *
+     * @since 3.2.9
+     *
+     * @param string $hook Stores the hook of the page.
+     *
+     * @see wp_enqueue_script function is relied on
+     * @link https://developer.wordpress.org/reference/functions/wp_enqueue_script/
+     */
+    public function settings_enqueue_script( $hook ) {
+      if ( 'settings_page_drift-settings' != $hook ) {
+        return;
+      }
+      wp_enqueue_script( 'drift-custom-js', $this->plugin_URL . '/admin/js/custom.min.js' );
+    }
+
+    /**
+     * Determines the hook.
+     *
+     * Determines the hook where Drift JavaScript will be loaded.
+     *
+     * @since 3.2.9
+     * @access private
+     *
+     * @see get_option function is relied on
+     * @link https://developer.wordpress.org/reference/functions/get_option/
+     *
+     * @return string wp_footer Drift JS will be loaded in the site footer (for page speed)
+     *                wp_head   Drift JS will be loaded in the site header (for quickly show live chat)
+     */
+    private function get_js_load_hook() {
+      $options = get_option( 'drift_options' );
+      return isset( $options['js_hook'] ) ? 'wp_footer' : 'wp_head';
+    }
+
+    /**
+     * Enqueues scripts in front pages.
+     *
+     * Enqueues the javascript from Drift options in pages and posts.
+     *
+     * @since 3.2.9
+     *
+     * @see get_option function is relied on
+     * @link https://developer.wordpress.org/reference/functions/get_option/
+     *
+     * @see get_the_ID function is relied on
+     * @link https://developer.wordpress.org/reference/functions/get_the_ID/
+     *
+     * @see is_feed function is relied on
+     * @link https://developer.wordpress.org/reference/functions/is_feed/
+     *
+     * @see is_robots function is relied on
+     * @link https://developer.wordpress.org/reference/functions/is_robots/
+     *
+     * @see is_trackback function is relied on
+     * @link https://developer.wordpress.org/reference/functions/is_trackback/
+     *
+     * @see is_single function is relied on
+     * @link https://developer.wordpress.org/reference/functions/get_the_ID/
+     *
+     * @see is_page function is relied on
+     * @link https://developer.wordpress.org/reference/functions/is_page/
+     */
+    public function enqueue_script() {
+      $options = get_option( 'drift_options' );
+      $this_page_id = get_the_ID();
+      if ( isset( $options['hideon_pages'] ) ) {
+        $hideon_pages = array_map( 'absint', explode( ',', $options['hideon_pages'] ) );
+      } else {
+        $hideon_pages = array();
+      }
+      if ( is_feed() || is_robots() || is_trackback() ||
```

Comment:
could we perhaps make this a private function? It's a bit hard to understand everything going on

Reply:
Hi @RollForReflex,
I'm affraid enqueue_script method cannot be private because it's used in the wp_footer/wp_head hook and it has to be public.